### ISSUE TITLE
win_ransomware_shadowcopy.yml can't detect grandcrab's wmic.exe shadowcopy delete

### ISSUE THREAD

**Issue by weiofcn** [SigmaHQ/sigma#435]
IOC MD5:97a910c50171124f2cd8cfc7a4f2fa4f
sysmon capture:
`CommandLine: "C:\Windows\system32\wbem\wmic.exe" shadowcopy delete`

sigmac result:
`./tools/sigmac -t es-qs -c winlogbeat rules/windows/process_creation/win_ransomware_shadowcopy.yml
winlog.event_data.CommandLine.keyword:(*vssadmin\ delete\ shadows* OR *wmic\ SHADOWCOPY\ DELETE*)
`

original DSL:
`
{
  "query": {
    "bool": {
      "must": [
        {
          "query_string": {
            "query": "winlog.event_data.CommandLine.keyword:(*vssadmin\\ delete\\ shadows* OR *wmic\\ SHADOWCOPY\\ DELETE)",
            "default_field": "*"
          }
        }
      ]
    }
  }
}
`

Working DSL:
`
{
  "query": {
    "bool": {
      "must": [
        {
          "query_string": {
            "query": "winlog.event_data.CommandLine.keyword:(*vssadmin* delete\\ shadows* OR *wmic* SHADOWCOPY\\ DELETE)",
            "default_field": "*"
          }
        }
      ]
    }
  }
}
`

Question:
1. I deleted  the double backslash in the query of Working DSL, and it still worked, what is the \\ suppose to do?
2. The ransomware(97a910c50171124f2cd8cfc7a4f2fa4f) call the ```CommandLine:"C:\Windows\system32\wbem\wmic.exe" shadowcopy delete```, which perfectly work around the original DSL, I try to modify the DSL and manually generate `Working DSL`, is it right way to solve the problem or is there a better one?

**Comment by thomaspatzke** [comment]
It's likely a bug in the dsl backend.

**Comment by thomaspatzke** [comment]
> It's likely a bug in the dsl backend.

I was wrong here...

The rule describes a match on `wmic SHADOWCOPY DELETE` somewhere in the command line and doesn't matches on the command line of this malware sample. So the rule has to be adapted accordingly such that it will match on `*wmic*SHADOWCOPY DELETE` or even something more tolerant against changed command lines. To answer your question:

>     1. I deleted  the double backslash in the query of Working DSL, and it still worked, what is the \ suppose to do?

A normal space would be interpreted as separator token. The backslash keeps `wmic` ans `SHADOWCOPY` together in one string, separated by a space that is escaped by the backslash.

>     2. The ransomware(97a910c50171124f2cd8cfc7a4f2fa4f) call the `CommandLine:"C:\Windows\system32\wbem\wmic.exe" shadowcopy delete`, which perfectly work around the original DSL, I try to modify the DSL and manually generate `Working DSL`, is it right way to solve the problem or is there a better one?

Generally it's better to change the Sigma rule instead of the generated query, which is also possible in this case. It should look as follows:

```
title: Ransomware Deletes Volume Shadow Copies
status: experimental
description: Detects commands that delete all local volume shadow copies as used by different Ransomware families
references:
    - https://www.bleepingcomputer.com/news/security/why-everyone-should-disable-vssadmin-exe-now/
    - https://www.hybrid-analysis.com/sample/ed01ebfbc9eb5bbea545af4d01bf5f1071661840480439c6e5babe8e080e41aa?environmentId=100
author: Florian Roth
date: 2019/06/01
logsource:
    category: process_creation
    product: windows
detection:
    selection:
        CommandLine:
            - '*vssadmin delete shadows*'
            - '*wmic* SHADOWCOPY DELETE*'
    condition: selection
fields:
    - CommandLine
    - ParentCommandLine
falsepositives:
    - Adminsitrative scripts - e.g. to prepare image for golden image creation
```

Can you test it with this rule and confirm if it works? Then we can change this rule.

By the way, there's a *es-dsl* backend that generates more ES DSLish queries, in this case:
```
{
  "query": {
    "constant_score": {
      "filter": {
        "bool": {
          "should": [
            {
              "wildcard": {
                "winlog.event_data.CommandLine.keyword": "*vssadmin delete shadows*"
              }
            },
            {
              "wildcard": {
                "winlog.event_data.CommandLine.keyword": "*wmic SHADOWCOPY DELETE*"
              }
            }
          ]
        }
      }
    }
  }
}
```

**Comment by weiofcn** [comment]
I found two defects in the `win_ransomware_shadowcopy.yml`

1. `"*vssadmin* delete shadows*"` and `"*wmic* SHADOWCOPY DELETE*"` work well, `vssadmin` and `vssadmin.exe` both work in cmdline, so `*` is needed, similarly, it happened on `wmic`.

2. `keyword` in ES is case sensitive, solves by `https://discuss.elastic.co/t/wildcard-case-insensitive-query-string/75050/4`, currently I did research on more than 10 ransoms, I found there are many ways in the cmdline to delete shadows, such as `vssadmin.exe  Delete Shadows /All /Quiet` in `globelmposter`, `vssadmin  delete shadows /all /quiet` in `crysis`, so make them lower case may important when query by DSL.
